dotCMS/src/main/webapp/html/js/dotcms/dojo/data/TemplateReadStore.js

```diff
@@ -13,188 +21,362 @@ dojo.declare("dotcms.dojo.data.TemplateReadStore", null, {
 	allSiteLabel: false,
 
 	/**
-	 * To Emulate this old Backend Behavion
+	 * To Emulate this old Backend Behavior
 	 * https://github.com/dotCMS/core/blob/7bc05d335b98ffb30d909de9ec82dd4557b37078/dotCMS/src/main/java/com/dotmarketing/portlets/templates/ajax/TemplateAjax.java#L72-L76
-	 * 
-	 * @type {*}
-	 * */
+	 */
 	ALL_SITE_TEMPLATE: {
 		title: "All Sites",
 		fullTitle: "All Sites",
 		htmlTitle: '<div>-- All Sites ---</div>',
 		identifier: "0",
-		inode: "0"
+		inode: "0",
+		hostName: "",
+		type: "template",
+		modDate: new Date().getTime()
 	},
 
 	constructor: function (options) {
-		this.hostId = options.hostId;
-		this.allSiteLabel = options.allSiteLabel;
-		this.templateSelected = options.templateSelected;
+		if (!options) {
+			throw new Error('TemplateReadStore: options parameter is required');
+		}
+
+		this.hostId = options.hostId || '';
+		this.allSiteLabel = !!options.allSiteLabel;
+		this.templateSelected = options.templateSelected || '';
+		this.DEBUG_MODE = !!options.debug;
+	},
+
+	// Utility Methods
+	_log: function(message, ...args) {
+		if (this.DEBUG_MODE) {
+			console.log(`[TemplateReadStore] ${message}`, ...args);
+		}
 	},
 
+	_error: function(message, error) {
+		console.error(`[TemplateReadStore] ${message}`, error);
+	},
+
+	_warn: function(message, ...args) {
+		console.warn(`[TemplateReadStore] ${message}`, ...args);
+	},
+
+	_validateResponse: function(responseEntity) {
+		if (!responseEntity || !Array.isArray(responseEntity.entity)) {
+			throw new Error('Invalid API response structure');
+		}
+		return true;
+	},
+
+	_buildApiUrl: function(query, pagination) {
```

Comment:
I suggest using `URL` and `searchParams` to build a URL properly, example for this case:

```ts
_buildApiUrl(query, pagination) {
	const { page, perPage } = pagination;
	const filter = query.fullTitle ? query.fullTitle.replace('*', '') : '';

	const url = new URL(this.API_BASE_URL);

	url.searchParams.set('filter', filter);
	url.searchParams.set('page', page);
	url.searchParams.set('per_page', perPage);

	if (query.sort) {
		url.searchParams.set('orderby', query.sort);
	}

	if (query.hostId) {
		url.searchParams.set('host', query.hostId);
	}

	return url.toString();
}
```